fbw-common/src/wasm/systems/systems/src/apu/pw980.rs

```diff
@@ -0,0 +1,1188 @@
+                    + (APU_FREQ_X6 * n.powi(6)),
+            )
+        } else {
+            Frequency::new::<hertz>(400.)
+        }
+    }
+
+    fn should_provide_output(&self) -> bool {
+        !self.failure.is_active()
+            && !self.is_emergency_shutdown
+            && self.n.get::<percent>() >= Pw980ApuGenerator::APU_GEN_POWERED_N
+    }
+}
+impl ApuGenerator for Pw980ApuGenerator {
+    fn update(&mut self, n: Ratio, is_emergency_shutdown: bool) {
+        self.n = n;
+        self.is_emergency_shutdown = is_emergency_shutdown;
+    }
+
+    /// Indicates if the provided electricity's potential and frequency
+    /// are within normal parameters. Use this to decide if the
+    /// generator contactor should close.
+    /// Load shouldn't be taken into account, as overloading causes an
+    /// overtemperature which over time will trigger a mechanical
+    /// disconnect of the generator.
+    fn output_within_normal_parameters(&self) -> bool {
+        self.should_provide_output() && self.potential_normal() && self.frequency_normal()
+    }
+}
+provide_potential!(Pw980ApuGenerator, (110.0..=120.0));
+provide_frequency!(Pw980ApuGenerator, (390.0..=410.0));
+provide_load!(Pw980ApuGenerator);
+impl ElectricalElement for Pw980ApuGenerator {
+    fn input_identifier(&self) -> ElectricalElementIdentifier {
+        self.identifier
+    }
+
+    fn output_identifier(&self) -> ElectricalElementIdentifier {
+        self.identifier
+    }
+
+    fn is_conductive(&self) -> bool {
+        true
+    }
+}
+impl ElectricitySource for Pw980ApuGenerator {
+    fn output_potential(&self) -> Potential {
+        if self.should_provide_output() {
+            Potential::new(
+                PotentialOrigin::ApuGenerator(self.number),
+                self.output_potential,
+            )
+        } else {
+            Potential::none()
+        }
+    }
+}
+impl SimulationElement for Pw980ApuGenerator {
+    fn accept<T: crate::simulation::SimulationElementVisitor>(&mut self, visitor: &mut T) {
+        self.failure.accept(visitor);
+        visitor.visit(self);
+    }
+
+    fn write(&self, writer: &mut SimulatorWriter) {
+        self.writer.write_alternating_with_load(self, writer);
+    }
+
+    fn process_power_consumption_report<T: PowerConsumptionReport>(
+        &mut self,
+        _: &UpdateContext,
+        report: &T,
+    ) {
+        self.output_potential = if self.should_provide_output() {
+            self.calculate_potential(self.n)
+        } else {
+            ElectricPotential::default()
+        };
+
+        self.output_frequency = if self.should_provide_output() {
+            self.calculate_frequency(self.n)
+        } else {
+            Frequency::default()
+        };
+
+        let power_consumption = report
+            .total_consumption_of(PotentialOrigin::ApuGenerator(self.number))
+            .get::<watt>();
+        let power_factor_correction = 0.8;
+        let maximum_load = 90000.;
```

Comment:
```suggestion
        let maximum_load = 120000.;
```
The APU generators are rated for 120KVA on the A380.